### ISSUE TITLE
Test plan for basic code intel

### ISSUE THREAD

**Issue by nicksnyder** [sourcegraph/sourcegraph#1712]
Please update this issue with a test plan and then assign testers. Close this issue when testing is finished.

**Comment by sqs** [comment]
## Screenshot 1

<details>
![screenshot from 2019-01-08 23-42-31](https://user-images.githubusercontent.com/1976/50884212-15530180-139f-11e9-94ee-b74f91250776.png)
</details>

## Screenshot 2

<details>
![screenshot from 2019-01-08 23-42-10](https://user-images.githubusercontent.com/1976/50884194-0b310300-139f-11e9-900e-9482bc26b99b.png)
</details>

## Screenshot 3

<details>
At https://sourcegraph.sgdev.org/github.com/sourcegraph/sourcegraph/-/blob/cmd/server/shared/shared.go#L66:6&tab=references

![screenshot from 2019-01-08 23-44-03](https://user-images.githubusercontent.com/1976/50884277-4cc1ae00-139f-11e9-8893-b8e547587a1e.png)
</details>

## Screenshot 4

<details>
![screenshot from 2019-01-08 23-44-42](https://user-images.githubusercontent.com/1976/50884297-619e4180-139f-11e9-998e-c9730543b425.png)
</details>

**Comment by lguychard** [comment]
## Screenshot 1

<details>

![image](https://user-images.githubusercontent.com/1741180/50897699-318d8780-140d-11e9-825c-0e53c74b9de9.png)

</details>

## Screenshot 2

<details>

![image](https://user-images.githubusercontent.com/1741180/50897740-55e96400-140d-11e9-8215-cc45d90eacf9.png)

</details>

## Screenshot 3

<details>
At https://sourcegraph.sgdev.org/github.com/gorilla/mux/-/blob/middleware_test.go#L25:9&tab=references

![image](https://user-images.githubusercontent.com/1741180/50897766-6994ca80-140d-11e9-8f7f-a0e63f8fc90b.png)

</details>

## Screenshot 4

<details>

![image](https://user-images.githubusercontent.com/1741180/50897845-a9f44880-140d-11e9-8c2a-1856432cfb4f.png)

</details>

## Support request responses

- "Why do I only see 'Find references' and not 'Go to definition' on my Scala code?"
    - The Sourcegraph basic-code-intel extension uses [universal-ctags](https://github.com/universal-ctags/ctags) to provide definitions. Not all languages are supported.
- "The extension is showing many incorrect references for my function named open."
    - Because basic-code-intel uses text-based heuristics, its definition and reference results are not precise. This is why it returns many results on commonly used tokens like "open".

**Comment by felixfbecker** [comment]
Which languages are expected to work?

**Comment by sqs** [comment]
@felixfbecker All languages.

**Comment by sqs** [comment]
Yes, it is. Just milestoned it. It actually exhibits unexpected behavior - it still should show the code line even if not the doc line, so I want to investigate.

**Comment by felixfbecker** [comment]
https://github.com/sourcegraph/sourcegraph/issues/1779

**Comment by felixfbecker** [comment]
Basic code intel is not working for me at all, not even on sourcegraph.com.

**Comment by sqs** [comment]
@felixfbecker Can you file an issue for basic code intel not working for you at all on Sourcegraph.com, and include all the relevant info?

I see your user settings have the following (https://sourcegraph.com/users/felixfbecker/settings):

```
  "basicCodeIntel.enabled": false,
```

So is it possible you disabled it previously? The command palette action `Basic code intel: Enable fuzzy def/ref matching` will reenable it.

- What troubleshooting did you attempt?
- Did you see the mention of this toggle action at https://sourcegraph.com/extensions/sourcegraph/basic-code-intel under **Optional features**?

**Comment by vanesa** [comment]
## Screenshot 1

<details>
![image](https://user-images.githubusercontent.com/9110008/51035668-50427880-15dd-11e9-9a9f-6c7233707568.png)

</details>

## Screenshot 2
I did experience #1738. First I could see `show definition` and after a few seconds it disappeared and only showed `find references`.
<details>
![image](https://user-images.githubusercontent.com/9110008/51041108-bdf5a100-15eb-11e9-9d00-b5184d966148.png)

![image](https://user-images.githubusercontent.com/9110008/51040953-593a4680-15eb-11e9-983d-d53eae609f3d.png)

</details>

## Screenshot 3

<details>

https://sourcegraph.sgdev.org/github.com/gorilla/mux/-/blob/mux.go#L175:18&tab=references
![image](https://user-images.githubusercontent.com/9110008/51041347-4f651300-15ec-11e9-96cf-ffb2b9de06e4.png)

</details>

(saving in between)

**Comment by sqs** [comment]
Thanks, @vanesa! @felixfbecker Can you run through the test plan now (see https://github.com/sourcegraph/sourcegraph/issues/1779#issuecomment-453413334 https://github.com/sourcegraph/sourcegraph/issues/1367#issuecomment-453639301)?

**Comment by sqs** [comment]
Between @lguychard 	and @vanesa, this already has 2 successful test runs, so I am removing the release-blocker label.

**Comment by nicksnyder** [comment]
Additional testing on this is not a priority for 3.0-beta, so moving to 3.0

**Comment by nicksnyder** [comment]
@sqs is any more testing needed here for 3.0?

**Comment by sqs** [comment]
Yes, this will need more testing because we are making it installed by default and multiple extensions. @chrismwendt owns that part of testing.

**Comment by sqs** [comment]
@chrismwendt please update this issue with modified test plan accounting for installed by default and multiple extensions.

**Comment by chrismwendt** [comment]
@sqs Done, I updated the description with the modified test plan.
